plugins/rapid7_insightidr/komand_rapid7_insightidr/util/resource_helper.py

```diff
@@ -64,3 +92,42 @@ def resource_request(self, endpoint: str, method: str = "get", params: dict = No
             status_code_message = self._ERRORS.get(response.status_code, self._ERRORS[000])
             self.logger.error(f"{status_code_message} ({response.status_code}): {error}")
             raise PluginException(f"InsightIDR returned a status code of {response.status_code}: {status_code_message}")
+
+    @staticmethod
+    async def _get_log_entries_with_labels(connection: Connection, log_entries: [dict]) -> [dict]:
```

Comment:
Please write a comment or two describing the functionality of this method, as it appears complex.
My understanding is that this method asynchronously collects new labels (by log ID) and merges the old `{log ID -> label}` dict object with newly acquired ID -> label key-value pairs.